### PR TITLE
MNT Fixing documentation

### PR DESCRIPTION
**Pull request by riedgar-ms**
## Description

Applying fixes for the documentation:

- In `plot_new_metrics.py` it appears that `fetch_adult()` is now returning something slightly different.... or pandas `map()` function is now including missing data

Note that this does not address the bounty of `pandas` warnings we're getting right now. Those warning don't kill this build (and should be addressed by #1294 )

## Tests

- [x] no new tests required
- [ ] new tests added
- [ ] existing tests adjusted

## Documentation

- [ ] no documentation changes needed
- [ ] user guide added or updated
- [ ] API docs added or updated
- [x] example notebook added or updated

### CODE REVIEW

File: examples/plot_new_metrics.py [hunk 1]
```diff
@@ -109,7 +109,11 @@ def marriage_transform(m_s_string):
 def occupation_transform(occ_string):
     """Perform some simple manipulations."""
     result = "Small"
-    if occ_string.startswith("Machine"):
+    if isinstance(occ_string, float):
+        # Had missing data
+        assert np.isnan(occ_string)
+        result = "Low"
```
Comment (hildeweerts) [inline examples/plot_new_metrics.py:115]: Should this say "Small"?

**Review by hildeweerts — COMMENTED** [review]
Could this be related to https://github.com/scikit-learn/scikit-learn/pull/26551?

File: examples/plot_new_metrics.py (hunk 1 above)
Comment (riedgar-ms) [reply examples/plot_new_metrics.py:115]: Yes. I must have read the wrong `fillna()` value below

Comment (riedgar-ms) [reply examples/plot_new_metrics.py:115]: I'll update the logic to correct this.

**Comment by riedgar-ms** [comment]
> Could this be related to [scikit-learn/scikit-learn#26551](https://github.com/scikit-learn/scikit-learn/pull/26551)?

Quite possibly. Alternatively, `DataFrame.map()` may have changed its behaviour, and now always passes 'missing' data to the underlying function.

**Review by hildeweerts — APPROVED** [review]
> Alternatively, DataFrame.map() may have changed its behaviour, and now always passes 'missing' data to the underlying function.

tbh this would surprise me, given that `map()` has an explicit argument to control how to deal with missing data and the default value has not changed since previous versions. In any case, changes are approved 👍

**Comment by riedgar-ms** [comment]
I missed that @hildeweerts . So we definitely blame SciKit-Learn, then :-)

**Pull request merged**